ddtrace/tracer/payload_v1.go

```diff
@@ -0,0 +1,1426 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
+	"github.com/DataDog/dd-trace-go/v2/internal/log"
+	"github.com/DataDog/dd-trace-go/v2/internal/processtags"
+	"github.com/tinylib/msgp/msgp"
+)
+
+// payloadV1 is a new version of a msgp payload that can be sent to the agent.
+// Be aware that payloadV1 follows the same rules and constraints as payloadV04. That is:
+//
+// payloadV1 is not safe for concurrent use
+//
+// payloadV1 is meant to be used only once and eventually dismissed with the
+// single exception of retrying failed flush attempts.
+//
+// ⚠️  Warning!
+//
+// The payloadV1 should not be reused for multiple sets of traces.  Resetting the
+// payloadV1 for re-use requires the transport to wait for the HTTP package
+// Close the request body before attempting to re-use it again!
+type payloadV1 struct {
+	// bm keeps track of which fields have been set in the payload
+	// bits 1-11 are used for field IDs 1-11. Bit 0 is unused.
+	bm bitmap
+
+	// the string ID of the container where the tracer is running
+	containerID string // 2
+
+	// the string language name of the tracer
+	languageName string // 3
+
+	// the string language version of the tracer
+	languageVersion string // 4
+
+	// the string version of the tracer
+	tracerVersion string // 5
+
+	// the V4 string UUID representation of a tracer session
+	runtimeID string // 6
+
+	// the optional `env` string tag that set with the tracer
+	env string // 7
+
+	// the optional string hostname of where the tracer is running
+	hostname string // 8
+
+	// the optional string `version` tag for the application set in the tracer
+	appVersion string // 9
+
+	// a collection of key to value pairs common in all `chunks`
+	attributes map[string]anyValue // 10
+
+	// a list of trace `chunks`
+	chunks []traceChunk // 11
+
+	// protocolVersion specifies the trace protocol to use.
+	protocolVersion float64
```

Comment:
Isn't `protocolVersion` for the `payloadV1` static -- always traceProtocolV1? So we shouldn't need a field to store it, the method can just return traceProtocolV1